src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -64,6 +65,53 @@ std::vector<std::string> get_supported_metrics() {
           "confusion_matrix", "roc_curve"};
 }
 
+size_t count_correct_predictions(size_t num_classes, const shared_float_array& output_chunk, 
+    const shared_float_array& label_chunk, size_t num_predictions) {
+      
+  const float* output_ptr = output_chunk.data();
+  const float* truth_ptr = label_chunk.data();
+    
+
+  size_t num_correct_predictions = 0.f;
+  
+  for (size_t i = 0; i < num_predictions; i++) {
+
+    size_t prediction = std::max_element(output_ptr, output_ptr + num_classes) - output_ptr;
+    if (prediction == *truth_ptr) {
+      num_correct_predictions += 1;
+      };
+
+    truth_ptr++;
+    output_ptr += num_classes;
+
+  }
+
+  return num_correct_predictions;
+
+}
+
+
+float get_accuracy_per_batch(size_t prediction_window, size_t num_classes, 
+  const shared_float_array& output, const data_iterator::batch& batch) {
```

Comment:
Here too we should indent two additional spaces to visually set this apart from the body's indent level